lib/core/src/Cardano/Wallet/Primitive/BlockSummary.hs

```diff
@@ -148,31 +155,32 @@ data BlockEvents = BlockEvents
 -- When processing whole 'Block', we want to avoid copying
 -- and redecorating the entire list of transactions in that 'Block';
 -- instead, we want to copy a pointer to this list.
-data Sublist a = All [a] | Some [(Int, a)]
+data Sublist a = All [a] | Some (Map (Index1, Index2) a)
```

Comment:
👍 We could have used a type parameter `Sublist ix a` and avoided the specialization to a pair, but meh — the pair is internal, and the `Sublist` encapsulation is good enough for this module.

(That said, sometimes the code becomes simpler by generalizing it in the right way — a common occurrence in mathematics. But meh.)